README.md

```diff
@@ -2,43 +2,21 @@
 [![Build Status](https://travis-ci.org/haskell/ghcup.svg?branch=master)](https://travis-ci.org/haskell/ghcup)
 [![license](https://img.shields.io/github/license/haskell/ghcup.svg)](COPYING)
 
-# GHC up
+Although [`cabal-install`](https://hackage.haskell.org/package/cabal-install) automatically installs Haskell library dependencies, and selects a specific version of `ghc` (see [`with-compiler`](https://cabal.readthedocs.io/en/latest/nix-local-build.html#cfg-flag---with-compiler)). However, `cabal-install` will fail if the required version of `ghc` is not available. Unfortunately, many operating systems do not offer a way to install specific versions of `ghc`.
```

Comment:
You removed a lot of information (e.g. where ghcup places ghc versions etc). I'm not sure this makes it any clearer? We can have some clearer intro sentence maybe, but I still want to be up-front to the users about what this does on the lower level.

Reply:
I can add this back, but it feels like implementation detail to me and should therefore be lower down after installation and usage.